src/yki/handler/registration.clj

```diff
@@ -1,59 +1,94 @@
-               (ok {:success true}))
-             (do
-               (log/error "Registration id:" id "failed with error" error)
-               (internal-server-error {:success false
-                                       :error
-                                       error})))))))))
+    (context routing/registration-api-root []
+      :coercion :spec
+      :middleware [auth access-log]
+      (POST "/init" request
+        :body [registration-init ::ys/registration-init]
+        :return ::ys/registration-init-response
+        (audit/log-participant {:request   request
+                                :target-kv {:k audit/registration-init
+                                            :v (:exam_session_id registration-init)}
+                                :change    {:type audit/create-op
+                                            :new  registration-init}})
+        (registration/init-registration db
+                                        (or user-config (:session request))
+                                        registration-init
+                                        (:payment-config payment-helper)))
+      (context "/:id" []
+        (POST "/submit" request
+          :body [registration ::ys/registration]
+          :path-params [id :- ::ys/id]
+          :query-params [lang :- ::ys/language-code
+                         {use-yki-ui :- ::ys/use-yki-ui nil}]
+          :return ::ys/response
+          (let [{:keys [oid error]} (registration/submit-registration db
+                                                                      url-helper
+                                                                      payment-helper
+                                                                      email-q
+                                                                      lang
+                                                                      (or user-config (:session request))
+                                                                      id
+                                                                      registration
+                                                                      onr-client
+                                                                      use-yki-ui)]
+            (if oid
+              (do
+                (audit/log-participant {:request   request
+                                        :oid       oid
+                                        :target-kv {:k audit/registration
+                                                    :v id}
+                                        :change    {:type audit/create-op
+                                                    :new  registration}})
+                (ok {:success true}))
+              (do
+                (log/error "Registration id:" id "failed with error" error)
+                (internal-server-error {:success false
+                                        :error   error})))))
+        (POST "/cancel" request
```

Comment:
En tiedä miten tarkkaan ykissä näitä on seurattu/mietitty. Teknisestihän sillä ei oikeastaan ole väliä vaan enemmän makuasia, mutta pitäisköhän tässä olla PUT kun kyseessä on päivitysoperaatio? 🤔